packages/sdk/src/base.ts

```diff
@@ -0,0 +1,34 @@
+import { listen } from './request';
+import { LogCallback } from './interface';
+
+export class BaseApi {
+  route: string;
+
+  constructor(uri: string) {
+    this.route = uri;
+  }
+
+  /**
+   * listen log
+   * @param logId log id
+   * @param logCallback log callback
+   */
+  async log(logId: string, logCallback: LogCallback) {
+    return new Promise((resolve, reject) => {
+      listen(`${this.route}/log/${logId}`, undefined, {
+        'log': (e: MessageEvent) => {
+          const logObj = JSON.parse(e.data);
+          if (logCallback) {
+            logCallback(logObj.level, logObj.data);
+          }
+        },
+        'error': (e: MessageEvent) => {
+            reject(e.data);
+        },
+        'close': () => {
```

Comment:
```suggestion
        'close': resolve
```

Reply:
The resolve type is `(value?: void | PromiseLike<void>) => void`, it's not suitable for the `EventListener`.